### PR TITLE
Testing rspec

### PR DESCRIPTION
**Pull request by balerum03**
**Project specific**

- Add specs for all public methods you created in Tic Tac Toe project.

- Use existing repositories for Tic Tac Toe and Enumerables: add specs and open new Pull Requests.

- Wrap specs for each method in describe block, hints.

- Instead of using before_each to define a variable for the test cases, use let.

- Use one it block for each expected behaviour. Do not test multiple aspects in one it block.

- Use your common sense: test positive and negative scenarios.

### CODE REVIEW

File: spec/tic_tac_toe_spec.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    expect(board_obj.swap_on_board(2, 5)).to eql(board_obj.board)
+  end
+end
+
+describe Board do
+  let(:board_obj) { Board.new(%w[1 2 3 4 5 6 7 8 9]) }
+  it 'Changes the possition given for "O" ' do
+    expect(board_obj.swap_on_board(5, 9)).to eql(board_obj.board)
+  end
+end
```
Comment (bertil291utn) [inline spec/tic_tac_toe_spec.rb]: ## Wrap in just one describe
- Specs for each method should be wrapped in one `describe` block. As in example here https://github.com/rubocop-hq/rspec-style-guide#empty-lines-around-examples . ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/describe.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))
  - [x] Please wrap in just one describe
```ruby
describe Board do
it 'a  message'
...
it 'another message'
end

```
*Note: remember to add describe board sentences they're in this file *

File: spec/tic_tac_toe_spec.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    expect(interface.display_board(test_array).class).to eql(String)
+  end
+end
+
+describe Board do
+  let(:board_obj) { Board.new(%w[1 2 3 4 5 6 7 8 9]) }
+  it 'Changes the possition given for "X" ' do
+    expect(board_obj.swap_on_board(2, 5)).to eql(board_obj.board)
+  end
+end
```
Comment (bertil291utn) [inline spec/tic_tac_toe_spec.rb]: ## Negative scenario

- Specs cover positive and negative scenarios  (screenshot  from  Pathwright)
Example: there is a spec for correct usage of the method [e.g. it "creates object if all required attributes are passed"]
and also there is spec for the wrong usage of a method  [e.g. it "does not create the object if required attributes are missing"].

  - [x] Thus, please add negative scenarios

Example:

```ruby
#positive scenario
 it 'Changes the position given for "X" ' do
#negative scenario
 it 'doesn\'t change the position given for "X" ' do
```

File: spec/tic_tac_toe_spec.rb (hunk 1 above)
Comment (bertil291utn) [inline spec/tic_tac_toe_spec.rb]: ## Negative scenario

- Specs cover positive and negative scenarios  (screenshot  from  Pathwright)
Example: there is a spec for correct usage of the method [e.g. it "creates object if all required attributes are passed"]
and also there is spec for the wrong usage of a method  [e.g. it "does not create the object if required attributes are missing"].

  - [x] Thus, please add negative scenarios

Example:

```ruby
#positive scenario
 it 'Changes the position given for "O" ' do
#negative scenario
 it 'doesn\'t change the position given for "O" ' do
```

File: spec/tic_tac_toe_spec.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+describe Players do
+  let(:player_name) { Players.new('Adan', 'Sam') }
+  context "Returns player's one name when number is even" do
+    it 'Returns the name of the player' do
+      expect(player_name.get_name(0)).to eql('Adan')
+    end
+    context "Returns player's two name when number is odd" do
+      it 'Returns the name of the player' do
+        expect(player_name.get_name(1)).to eql('Sam')
+      end
```
Comment (bertil291utn) [inline spec/tic_tac_toe_spec.rb:62]: ## Negative scenario

- Specs cover positive and negative scenarios  (screenshot  from  Pathwright)
Example: there is a spec for correct usage of the method [e.g. it "creates object if all required attributes are passed"]
and also there is spec for the wrong usage of a method  [e.g. it "does not create the object if required attributes are missing"].

  - [x] Thus, please add negative scenarios

Example:

```ruby
#positive scenario
 context "Returns player's one name when a number is even" do
#negative scenario
context "doesn't return player's one name when a number is even" do```

File: spec/tic_tac_toe_spec.rb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    expect(board_obj.draw?).to eql(true)
+  end
+end
+
+describe Board do
+  let(:board_obj) { Board.new(%w[X O X X O X O X 9]) }
+  it 'Returns true when the array has no integers left on it' do
+    expect(board_obj.draw?).to eql(false)
+  end
+end
```
Comment (bertil291utn) [inline spec/tic_tac_toe_spec.rb]: ## Change message
- [x] Change the message 
```ruby
#from
 it 'Returns true when the array has no integers left on it' do
#to
 it 'Returns false when the array has no integers left on it' do
```

File: spec/tic_tac_toe_spec.rb [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  it 'Returns True when there is a winner otherwise it returns False' do
+    expect(board_obj.winner?(6)).to eql(true)
+  end
+end
+
+describe Board do
+  let(:board_obj) { Board.new(%w[X 2 3 4 5 6 6 8 9]) }
+  it 'Returns True when there is a winner otherwise it returns False' do
+    expect(board_obj.winner?(0)).to eql(false)
+  end
```
Comment (bertil291utn) [inline spec/tic_tac_toe_spec.rb]: ## Confusing message
- [ ] Please edit the confusing message
 - Edit like so
```ruby
# The first one
it 'Returns True when there is a winner ' do
# and the other one
it 'Returns false when there is no a winner ' do
```

**Review by bertil291utn — CHANGES_REQUESTED** [review]
# **Status:** Changes requested 🟠 
<p align="center"> image <br><strong>Not yet</strong> </p>

# Hey guys, you've been doing a great job, and also I hope you enjoyed coding.

### Your work is amazing you master it GitHub Flow you have a professional README file. 

However, in order to accomplish all the requirements , you required to do these changes, listed below:

- [x] Please instructions to run tests in the README file
```markdown
## Test
- All steps you need to describe to run the tests
```
- [x] Add negative scenarios. Check inline comments

### Summary
- Please add negative scenarios 
-  Don't forget to work with inline comments below this comment box.

***

Guys work in the requested changes above and request another review.
Good luck. 

> “Never stop, never conform until what is good becomes even better, and what’s better becomes excellent.”

**Review by tirthajyoti-ghosh — APPROVED** [review]
# Approved ✅

## You have done an incredible job making this project 👏🧨💪! Just perfect 💯! 

### You have met all the requirements of the project. Go ahead and merge it! Onto the next project 🚀 !

Happy coding 👨‍💻 

<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Pull request merged**